frontend/lib/src/components/transaction_component/transaction_component.dart

```diff
@@ -0,0 +1,178 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_form_builder/flutter_form_builder.dart';
+import 'package:form_builder_validators/form_builder_validators.dart';
+import 'package:frontend/src/components/input/input_field.dart';
+import 'package:get/get.dart';
+
+/*
+  Reusable transaction component for deposit, withdraw and transfer cash
+
+  @param label, String value to display what kind of transaction it is, 
+    basicaly something like a title.
+
+  @param transfer, bool value to tell if the transaction is transfering money.
+     to activate the email feild.
+*/
+
+class TransactionComponent extends StatefulWidget {
+  const TransactionComponent({
+    super.key,
+    required this.label,
+    required this.transfer,
```

Comment:
Instead of transfer, we can use **type**. it makes more sense that way. It is more flexible too if we want to add different fields to each transactions.

types can be: DEPOSIT, WITHDRAWAL, TRANSFER

Don't make it required too, you can assign a default value.